### ISSUE TITLE
10/2 Task MVP of LucidChart and GitHub 

### ISSUE THREAD

**Issue by ritahitching** [lzim/teampsd#1398]
**MVP of LucidChart and GitHub** 
@ritahitching 
@lzim  please confirm I got your task correctly:

- [ ] Test ability to build on existing LucidChart knowledge by adding integration to GitHub forked TeamPSD repo
- [ ] Think of the average use - don't use pro account
- [ ] Test if I can teach my self - link GitHub forked repo and LucidChart 

Also consolidate the persona details.

**Comment by ritahitching** [comment]
# Update on MVP - GitHub and LucidChart Integration
@lzim 

## Using Free Account
1. Set up a basic / free account just to test out mvp
2. Went to market place found GitHub app integration
3. Prompted to sign in to GitHub account
4. Can only use my own repos 
5. Unable to sync to LZ teampsd or mtl
6. Use forked ritahitching/teampsd_hitching_sandbox - although out of date
7. Received confirmation that GitHub and LucidChart synced 
8. Unable to not go any further without paying for a pro account
￼
![image](https://user-images.githubusercontent.com/54862187/81630227-e21cea00-93b9-11ea-9d73-175c507f54bb.png)

![image](https://user-images.githubusercontent.com/54862187/81629917-23f96080-93b9-11ea-9f97-e899cfadef7d.png)

![image](https://user-images.githubusercontent.com/54862187/81629930-2bb90500-93b9-11ea-9d8f-bcdc0fdf58a5.png)

**Comment by ritahitching** [comment]
@lzim Update:
- [ ] I integrated GitHub into TeamPSD Lucidchart using @lzim credentials 
- [ ] I was able  integrate GitHub and Lucid chart - see [here](https://github.com/ritahitching/hitching_sandbox_training/issues/1)
- [ ] The user needs to be within own repo to be able to paste the published link provided by LucidChart into an issue / pull request on Github for the LucidChart map to display
- [ ] I was not able to generate a new issue with the forked TeamPSD repo - and as such unable to link a 'live' LucidChart document within an issue

# Next Steps TeamPSD Repo:
- [ ] Integrate TeamPSD repo with GitHub  - need @lzim GitHub credentials or
- [ ] @lzim can follow the steps below and integrate directly

## Step 1
Log in to GitHub
## Step 2 & 3 Got to Market Place & Search for LucidChart
![image](https://user-images.githubusercontent.com/54862187/81884965-2f31c500-954e-11ea-913d-9c91c33c5155.png)
## Step 4 Select Configure Access
![image](https://user-images.githubusercontent.com/54862187/81885110-791aab00-954e-11ea-9fbb-d06e5b26f40c.png)
## Step 5 Authenticate GitHub login details
## Step 6 Insert a Published LucidChart document link into any GitHub Issue or Pull request
- **Open** a LucidChart document of interest
- Select **Share**
![image](https://user-images.githubusercontent.com/54862187/81885409-27beeb80-954f-11ea-9f3d-f20938c39836.png)
- Select **Advanced**
- Select **Publish**
- Select **Generate Publish Link**
- Copy **Published Link** by selecting the **blue paperclip** and paste link into GitHub Issue
![image](https://user-images.githubusercontent.com/54862187/81885488-458c5080-954f-11ea-9412-38aa94e94c94.png)
![image](https://user-images.githubusercontent.com/54862187/81885620-9603ae00-954f-11ea-8e65-74ed79cf3251.png)

**Comment by ritahitching** [comment]
https://app.lucidchart.com/documents/view/812cb945-f81e-4178-bbb4-b0a2dd8ad3dc/tHc0TqO34ce3

**Comment by ritahitching** [comment]
https://app.lucidchart.com/documents/view/fa2b9266-2561-426b-9008-408aa46a3969

**Comment by ritahitching** [comment]
https://app.lucidchart.com/documents/view/7e6f6c3f-b276-4d32-b6af-8ff1194d5a3a

**Comment by ritahitching** [comment]
https://app.lucidchart.com/documents/view/7e6f6c3f-b276-4d32-b6af-8ff1194d5a3a

**Comment by lzim** [comment]
@ritahitching 

The main things missing here from our **Team PSD 2.0 Design Thinking** instructions are **clear user-centered hypotheses guiding each step of MVP learning/development to avoid rework** (our _pareto_ or _80/20 principle_). 
- _What assumptions about **user pain points** are you testing with each prototyping step related to Lucid Chart and GitHub integration?_

A **mental model** shift will help: This is not a “task,” this is your part of an _iterative design process across the team, and you’re working on  this in parallel with everyone else’s **distributed** prototyping work._

Check out the **Teams > Design > Persona and Prototyping Wiki** instructions and **develop/refine your user hypotheses:**

 - Your **prototype is team continuous integration of Lucid Charts and GitHub to improve the Team PSD 2.0 workflow.**

- empathy <> define <> ideate <> prototype <> test

Here are some of **assumptions** that I see you made already at this point of your prototype that should have been checked out with users before proceeding to avoid rework:

1. We definitely do not want these **non-PSD style** maps on GitHub!  
	- The major team pain point is getting the **Lucid** <> **GitHub** continuous integration standardized and accessible for all Team PSD users to contribute _themselves_).
	- We need to **scale** Lucid contributions, **but folks can’t tell how to do so.** It’s frustrating and leads to re-work!
		- **User POV Lucid Chart Scale Problems we’ve discussed:** 
			- Multiple templates, 
			- Inconsistent naming conventions and organization (this will be more and more problematic with links),
			- Connector symbol mistakes on the account, 
			- Use of the non-brand dark red remains, etc. that 
**all needs to be cleaned up, to make sure anyone can create, edit/update a Lucid Charts flow map asynchronously at scale.**)

2. There is also **a significant problem:** Keeping our **_persona_** resources **private:**
	- Remove these links immediately!

Also, use your **GitHub** branches to **_prototype_** with other users. 
	- **Prototypes are handled with pull-requests to other users repos.** 
	- _Which users are testing asynchronously and in parallel, whether they can use your instructions to establish bidirectional GitHub <> Lucid Charts?_
	- **Jane and Debbie** need to be oriented next week on this, _have you prototyped enough with the existing **user persona library** to be ready to expand to these users?_

3. Since we want to make sure that all users can update their Lucid Charts in a style consistent way **your prototyping must account for user learning:**
	- Review your **concurrent video think aloud protocol observation** to refine user instructions for doing so are clear
	- Ideally since we’re at the end of week 2, you aim to be clear about the **80/20 approach that the team will learn over the new couple weeks, so that everyone can make an effective, continuous integrated GitHub/Lucid Chart contributions themselves.**

Keeping **Team PSD 2.0** folks in the loop about this prototype:
@lijenn @anthonycpichardo @anazariz @jamesmrollins @staceypark 

I’m excited, I think for many folks we are going to be very clear about our 80/20 today, as we’d hoped! 🏁

**Comment by ritahitching** [comment]
@ritahitching  updates [here](https://github.com/ritahitching/teampsd_hitching_sandbox/blob/master/resources/design/lucidchart/lucidchart_github_mvp.md)

**Comment by ritahitching** [comment]
Detailed Documentation Second Stories 


 

Do you see any places where moving forward independently could have been more aligned with the team or integrated better when you were done? 

Yes, I should have check in earlier in the map process. I wanted to present the work finalized, and would have saved time and effort and solution by that was better.  It’s ok to show a work in process, fearful in the past due to interpretation of feedback as critique rather than useful guidance.  Was reluctant to check early as did not want to interrupt.
 

Could a dependency or root assumption have been caught any sooner for the benefit of the team? 

That the integration would just work for everyone based on the fact that it was a ‘connector’. es, asking someone to paste a link of a map and seeing it is slow to open and inconsistent. 
 

Have you discovered a new empathic understanding of the situation from another users' POV (point of view)? 

Yes, rules and structure are a sign of respect and kindness for `another person. Feedback is something to be grateful of.  Be open and grateful for it.  Receive it with gratitude. Provide detailed feedback as a sign of my value of generosity.  Discern what I need from another person - make it easy for them to help me, and what time line I need.

**Comment by ritahitching** [comment]
Updated on MVP

1.  LucidChart Account holder and the GitHub Account Repo have to be the same
2. Lucid Chart functions on GitHubby way of pull requests or issues, not markdown files.

**Organization of LucidChart**
Feedback from users recommending organizing LucidChart into folders
 
Stacey: I like the idea of the workgroup folders as well, especially since our lzim/teampsd GitHub repository is organized the same way! It'll help with keeping things parallel and consistent. Thanks for checking out that idea  

**Flowchart Guides Requested by Users**
Jessilyn & Swap 
Guide within the Resources folder on LucidChart with basic instructions on how to make a flowmap would be helpful.

**GitHub & LucidChart Code Integration**
Anthony confirmed that Bookdown does not have the option to use embedded html code for flowmaps from Lucidchart - video here https://youtu.be/qzQfu_kveYE

**Comment by ritahitching** [comment]
**Comparison of instructions**
Feedback from users that compare guides for webpage for integration of LucidChart & GitHub https://lucidchart.zendesk.com/hc/en-us/articles/360021365711-The-Lucidchart-Connector-for-GitHub OR https://www.lucidchart.com/pages/integrations/github

James: Reviewed the two links - they are very consistent with the guidance you had provided for loading the APP into my GitHub repository.  I don't think they add anything materially to what you have already provided.  I believe **your instructions were clear.** 
**APP itself took about 24 hours to nest** into GitHub and begin working (it worked fine the next day).

**Comment by ritahitching** [comment]
**Generating a Preset Library of Colors  Shapes for Flowmaps**
    
Feedback from users:
Jess:  **Explanation** of meaning of flowmap shapes to be incorporate into a template would be useful.
Basic principles guide:
https://www.lucidchart.com/pages/flowchart-symbols-meaning-explained

**Video** on how to create first diagram helpful
https://lucidchart.zendesk.com/hc/en-us/articles/115005888706-Video-Create-your-first-document

**Comment by ritahitching** [comment]
**Feedback from New MVP Templates**
- Most recent iteration of the LucidChart MVP which is summarized as a flow map TeamPSD MVP https://app.lucidchart.com/documents/view/aa7dfa66-4d3d-43df-b350-db1a349f9a8d 
- Commensurate  **How to make a flowmap** 
https://app.lucidchart.com/documents/view/72f63de5-f3aa-453a-bbe8-a137e980836f
- Template **TeamPSD Core Flowmap Template** https://app.lucidchart.com/invitations/accept/acc2b45c-2fb9-40ca-a897-6bb4c61fb275

**Comment by ritahitching** [comment]
Repo Documentation

Main Folder [LucidChart](https://github.com/ritahitching/teampsd_hitching_sandbox/tree/master/resources/design/lucidchart)

[LucidChartConnector](https://github.com/ritahitching/teampsd_hitching_sandbox/tree/master/resources/design/lucidchart/how_to_add_lucidchart_connector)

[LucidChart Templates] (https://github.com/ritahitching/teampsd_hitching_sandbox/blob/master/resources/design/lucidchart/templates_lucidchart.md)

[LucidChart Hypothesis](https://github.com/ritahitching/teampsd_hitching_sandbox/blob/master/resources/design/lucidchart/lucidchart_github_hypothesis.md)

[LucidChart Issues](https://github.com/ritahitching/teampsd_hitching_sandbox/blob/master/resources/design/lucidchart/issues_questions.md)

**Comment by lzim** [comment]
What week of **oct_epic** is this?

@staceypark @lijenn 

Are we running into capacity constraints?

**Comment by lijenn** [comment]
This was done last week, closing as the app is already integrated within GitHub and documentation in the manual.
